packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -129,13 +130,56 @@ class CodeSizeController {
     changeOldDiffSnapshotFile(oldFile);
     changeNewDiffSnapshotFile(newFile);
 
-    final diffMap = buildComparisonTreemap(oldFile.data, newFile.data);
+    Map<String, dynamic> diffMap;
+    if (oldFile.isApkFile && newFile.isApkFile) {
+      final oldApkProgramInfo = ProgramInfo();
+      _apkJsonToProgramInfo(
+        oldApkProgramInfo,
+        oldApkProgramInfo.root,
+        oldFile.data,
+      );
+
+      final newApkProgramInfo = ProgramInfo();
+      _apkJsonToProgramInfo(
+        newApkProgramInfo,
+        newApkProgramInfo.root,
+        newFile.data,
+      );
+
+      diffMap = compareProgramInfo(oldApkProgramInfo, newApkProgramInfo);
+    } else {
+      diffMap = buildComparisonTreemap(oldFile.data, newFile.data);
+    }
+
     diffMap['n'] = 'Root';
     final newRoot = generateDiffTree(diffMap);
 
     changeDiffRoot(newRoot);
   }
 
+  ProgramInfoNode _apkJsonToProgramInfo(
+    ProgramInfo program,
+    ProgramInfoNode parent,
+    Map<String, dynamic> json,
```

Comment:
make these required named parameters so the code reads easier

Reply:
Done

Reply:
Done